### PR TITLE
Improve k6 perf test results consistency

### LINKED ISSUE #8596: Inconsistent perf test results
### Description

The k6 perf test results fluctuate a lot across different runs for release 0.107. Close examination shows  it's largely due to k8s schedule dynamics that in every run, the speed to ramp up pods, and the actual pod CPU usage level vary a lot. Thus, we need to adjust resources configuration and HPA to get consistent resources thus consistent test results.

### Steps to reproduce

Run the k6 test suite multiple times and compare the results.

### Additional context

_No response_

### Hedera network

other

### Version

v0.107.1

### Operating system

None

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

This PR makes k8s deployment resources changes for better k6 perf tests consistency

- Set cpu resources requests to limits for rest, restjava, and web3
- Set HPA target CPU utilization to 40% for rest, restjava, and web3 

**Related issue(s)**:

Fixes #8596 

**Notes for reviewer**:

The changes have been tested and the results are in notion for v0.107.0-rc1, the `#2` run. Note most RPS are better than that of v0.106.0-rc1, some web3 tests showed 3 times higher RPS.

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by jnels124 — APPROVED** [review]
LGTM

**Review by jascks — COMMENTED** [review]
LGTM

**Review by jascks — APPROVED** [review]
LGTM again.

File: clusters/mainnet-staging-na/mainnet/helmrelease.yaml [hunk 1]
```diff
@@ -62,6 +62,29 @@ spec:
...
+        metrics:
+          - type: Resource
+            resource:
+              name: cpu
+              target:
+                type: Utilization
+                averageUtilization: 40
+      resources:
+        requests:
+          cpu: 1100m  # same as limits
```
[the remark this answers is not in the document]
Comment (xin-hedera) [reply clusters/mainnet-staging-na/mainnet/helmrelease.yaml:75]: Think I have different opinion. One of the main purposes of perf test is to find if we have regression or not. For that, we need a stable environment to reproduce the perf numbers as close as we can. 

For production envs, we don't have to provision it to be the same until we see increasing requests. In the long run, and in the extreme case, we see the same level of requests in production environment, we will have to adjust the resources similarly.

[the remark this answers is not in the document]
Comment (steven-sheehy) [reply clusters/mainnet-staging-na/mainnet/helmrelease.yaml:75]: Sure, think these are two schools of thought. We can go with your approach for now.

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**